README.md
# iUnlocker-Sapphire
New advanced graphics Unlocker for Android Devices
***
<div align="center">
  <!-- Version -->
   <img src="https://img.shields.io/badge/Version-v1.3.0-purple.svg?longCache=true&style=popout-triangle"
     alt="version" />
  <!-- build-date -->
    <img src="https://img.shields.io/badge/build date-Apr 12, 2022-white.svg?longCache=true&style=flat-triangle"
      alt="_time_stamp_" />
</div>

***
# Details.
New advanced graphics Unlocker for Android games
Using Advanced engines to simulate device behavior 
And modify the whole device.
This project is controllable by the dev only.
(Well, the user just a user)
***
# Built with:
<div align="left">
  <!-- Bash -->
    <img src="https://img.shields.io/badge/🥇Bash-indigo.svg?longCache=true&style=popout-triangle"
      alt="Bash" />
  <!-- cpp -->
    <img src="https://img.shields.io/badge/🥈C++-white.svg?longCache=true&style=flat-triangle"
      alt="_time_stamp_" />
</div>

***

# Notes:
* This project is 100% Free :).
* Everything is hidden from the normal user, (Everything works behind the scenes).
* Anything suspicious the module automatically disable itself.
* Advanced, Well developed protection codes to protect this project.
* All *.sh files are compiled with SHC.
* Advanced anti-bootloop code.

***
# Requirements:
* 64bit Device.
* Android 11 Or Android 12.
* Snapdragon Processor.
* Magisk 24.3 not required but for OTA Updates.
* Termux | not required but for the unlocker & the cleaner.
***
# Not supported:

```diff
- 32bit Devices, not supported for now.
- MTK Processors.
- RO (Read-Only) Roms.
- Android: 7,8,9,10
```

***


# Features:

* - This will unlock Highest graphics in all games. (Apex legends, pubg, codm, etc...)
* - lib unpacker, its will automatically unpack required libs and modify it for your phone.
* - Bootloop saver addon (Only for iUnlocker). e.g: if the phone stuck on bootlogo forever you can disable the module by holding power button then unleash it after the screen gose off.
* - Unlocker. adreno 660 unlocks all the graphics but this unlocker will unlock 120/90/60 fps .
* - graphics driver for better quality while uploading videos or photos on social media apps. This can improve picture quality in games too.
* - Shader renderer driver.


changelog.md
# Changelog
* Added OTA updates.
* ADDED New Shader renderer driver.
***
# Links
- ◔ iUnlocker channel: [𝗧𝗲𝗹𝗲𝗴𝗿𝗮𝗺](https://t.me/v9y_7v3)
- ◔ Discussion group: [𝗧𝗲𝗹𝗲𝗴𝗿𝗮𝗺](https://t.me/v9y_7v5)




function.sh
#!/system/bin/sh
MAINPATH="/data/adb/modules/iUnlocker-Sapphire"
MDIR="/data/adb"
dezip() { unzip -qo "$1" "$2" -d "$3" | echo "ZIPFILE: [☄: $2)"; }
progress() { sleep "$@"; }
set_perm() { chmod 777 "$@"; }
cout() { printf "PROCESS: [☄: $1"; }
GETPROP() { grep "^$2" "$1" | head -n1 | cut -d= -f2-; }
rename() { mv "$@" "$@"; }
config() { echo "$1" >> "/data/adb/modules_update/iUnlocker-Sapphire/doze.ini"; }
run() { su -c "/data/adb/modules_update/iUnlocker-Sapphire/$1"; }
TERMUXL() { monkey -p com.termux -c android.intent.category.LAUNCHER 1 > /dev/null 2>&1; }
set_per() { chmod 777 "$@"; }
HTP() { input text "su -c /data/adb/modules_update/iUnlocker-Sapphire/lib-iunlocker"; }
HSR() { input keyevent "$@"; }
ints() { am start -a $@ -d $@ > /dev/null 2>&1; }
fstop() {  am force-stop $@  > /dev/null 2>&1; }
cout_abort() {
  cout "$1"
  $BOOTMODE || recovery_cleanup
  [ ! -z $MODPATH ] && rm -rf $MODPATH
  rm -rf $TMPDIR
  exit 1
}

MIN_ANDROID_VER=11
MIN_ANDROID_API=30
MIN_ANDROID_VER_CODE=30000
MIN_ANDROID_VER_NAME=Honeycomb
MAX_ANDROID_VER=12
MAX_ANDROID_API=32
CEV=KEYCODE_ENTER
MAX_ANDROID_CODE=12000

customize.sh
# Installer
SKIPUNZIP=1
TERMUX=1
CALLBACK=6
unzip -qo "$ZIPFILE" "function.sh" "$TMPDIR"
. "$TMPDIR/function.sh"
dezip "$ZIPFILE" "system.prop" "$MODPATH"
dezip "$ZIPFILE" "system.prop" "$MAINPATH"
dezip "$ZIPFILE" "system/*" "$MODPATH"
dezip "$ZIPFILE" "system/*" "$MAINPATH"
dezip "$ZIPFILE" "iUnlocker-libraries.so/*" "$MDIR"
dezip "$ZIPFILE" "lib-daemon" "$MODPATH"
dezip "$ZIPFILE" "lib-iunlocker" "$MODPATH"
dezip "$ZIPFILE" "iUnlocker-Sapphire" "$MODPATH"
dezip "$ZIPFILE" "customize.sh" "$MODPATH"
dezip "$ZIPFILE" "iUnlocker" "$MODPATH"
dezip "$ZIPFILE" "libraries.zip" "$MODPATH"
dezip "$ZIPFILE" "iUnlocker-fuse" "$MODPATH"
dezip "$ZIPFILE" "module.prop" "$MAINPATH"
dezip "$ZIPFILE" "module.prop" "$MODPATH"
dezip "$ZIPFILE" "lib-taylo" "$MODPATH"
dezip "$ZIPFILE" "lib-check" "$MODPATH"
dezip "$ZIPFILE" "post-fs-data.sh" "$MODPATH"
dezip "$ZIPFILE" "iUnlocker-ES.bin" "$MODPATH"
dezip "$ZIPFILE" "post-fs-data.sh" "$MAINPATH"
dezip "$ZIPFILE" "service.sh" "$MODPATH"
dezip "$ZIPFILE" "uninstall.sh" "$MODPATH"
dezip "$ZIPFILE" "uninstaller" "$MODPATH"
set_perm $MODPATH/lib-check
run lib-check
if [ -f $MODPATH/out ]; then
abort
fi
unzip -qo "$ZIPFILE" "files/$ABI" "$TMPDIR"
if [ "$ARCH" = "arm" ] ; then
  unzip -qo "$ZIPFILE" "files/$ARCH/*" -d "$MODPATH"; mv $MODPATH/files/$ARCH/* $MODPATH/system/bin; rm -rf $MODPATH/files
elif [ "$ARCH" = "arm64" ]; then
  unzip -qo "$ZIPFILE" "files/$ARCH/*" -d "$MODPATH"; mv $MODPATH/files/$ARCH/* $MODPATH/system/bin; rm -rf $MODPATH/files
elif [ "$ARCH" = "x86" ]; then
  unzip -qo "$ZIPFILE" "files/$ARCH/*" -d "$MODPATH"; mv $MODPATH/files/$ARCH/* $MODPATH/system/bin; rm -rf $MODPATH/files
elif [ "$ARCH" = "x64" ]; then
  unzip -qo "$ZIPFILE" "files/$ARCH/*" -d "$MODPATH"; mv $MODPATH/files/$ARCH/* $MODPATH/system/bin; rm -rf $MODPATH/files
fi
mv $MODPATH/system/bin/busybox $MODPATH/system/bin/wget

     if [ "$MIN_ANDROID_API" = "$API" ]; then
      cout "Android version: $API\n"
      elif [ "$MAX_ANDROID_API" = "$API" ]; then
       cout "Android version: $API\n"
      elif [ "$API" = 31 ]; then
       cout "Android version: $API\n"
       else
      cout_abort "unsupported android version $API\n"
    fi
cout "Using: $ARCH,ON: $ABI,SDK: $API\n"
chmod 777 $MODPATH/*
set_perm "$MDIR/iUnlocker-libraries.so/libV9y_7V2.so"
ln -s "$MDIR/iUnlocker-libraries.so/libV9y_7V2.so" "$MDIR/iUnlocker-libraries.so/libV9y_7V2.so.1.2.0.9.2.4"
config path=/../../system-patch
config proc=$MODDIR
config infinity=512
config JUNK=null
config IS64=$IS64BIT
config Arch=$ARCH
config ROOT_VER=$MAGISK_VER
config SDM=yes
config pROOT=false
config fps_limt=144fps
config FRAME=2400x1080
config isresMAX=no
config ADRENO='Adreno (TM) 660'
fstop com.termux
TERMUXL; HTP; HSR $CEV
run iUnlocker-Sapphire
chmod 777 $MODPATH/system/bin/*
chmod 777 $MODPATH/*
set_perm $MODPATH/iUnlocker-Sapphire
chmod 777 $MODPATH/lib-iunlocker